magicctapipe/scripts/lst1_magic_real/lst1_magic_mc_dl0_to_dl1.py

```diff
@@ -1,248 +1,402 @@
+--input-file "./data/dl0/gamma_40deg_90deg_run1___cta-prod5-lapalma_LST-1_MAGIC_desert-2158m_mono_off0.4.simtel.gz"
+--output-file "./data/dl1/dl1_lst1_magic_gamma_40deg_90deg_off0.4_run1.h5"
+--config-file "./config.yaml"
+"""
 
 import time
 import yaml
+import logging
 import argparse
 import warnings
 import numpy as np
-from traitlets.config import Config
 from astropy import units as u
-from ctapipe.io import EventSource
-from ctapipe.io import HDF5TableWriter
-from ctapipe.core.container import Container, Field
+from astropy.coordinates import AltAz, SkyCoord, angular_separation
+from traitlets.config import Config
+from ctapipe.io import EventSource, HDF5TableWriter
+from ctapipe.core import Container, Field
 from ctapipe.calib import CameraCalibrator
-from ctapipe.image import ImageExtractor, hillas_parameters, leakage_parameters
-from ctapipe.image.cleaning import tailcuts_clean
-from ctapipe.image.timing import timing_parameters
-from ctapipe.image.morphology import number_of_islands
-from magicctapipe.utils import MAGIC_Cleaning, calc_impact
+from ctapipe.image import (
+    ImageExtractor,
+    tailcuts_clean,
+    apply_time_delta_cleaning,
+    number_of_islands,
+    hillas_parameters, 
+    timing_parameters, 
+    leakage_parameters
+)
+from ctapipe.coordinates import CameraFrame, TelescopeFrame
+from magicctapipe.utils import (
+    add_noise_in_pixels,
+    set_numba_seed,
+    random_psf_smearer,
+    apply_dynamic_cleaning,
+    MAGIC_Cleaning,
+    calc_impact
+)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
 
 warnings.simplefilter('ignore')
 
 __all__ = ['mc_dl0_to_dl1']
 
 
-class InfoContainer(Container):
+class EventInfoContainer(Container):
 
     obs_id = Field(-1, 'Observation ID')
     event_id = Field(-1, 'Event ID')
     tel_id = Field(-1, 'Telescope ID')
-    mc_energy = Field(-1, 'MC event energy', unit=u.TeV)
-    mc_alt = Field(-1, 'MC event altitude', unit=u.rad)
-    mc_az = Field(-1, 'MC event azimuth', unit=u.rad)
-    mc_core_x = Field(-1, 'MC Core X', unit=u.m)
-    mc_core_y = Field(-1, 'MC Core Y', unit=u.m)
-    mc_impact = Field(-1, 'MC Impact', unit=u.m)
-    alt_tel = Field(-1, 'Telescope altitude', unit=u.rad)
-    az_tel = Field(-1, 'Telescope azimuth', unit=u.rad)
+    alt_tel = Field(-1, 'Telescope altitude', u.deg)
+    az_tel = Field(-1, 'Telescope azimuth', u.deg)
+    mc_energy = Field(-1, 'MC event energy', u.TeV)
+    mc_alt = Field(-1, 'MC event altitude', u.deg)
+    mc_az = Field(-1, 'MC event azimuth', u.deg)
+    mc_disp = Field(-1, 'MC event disp', u.deg)
+    mc_core_x = Field(-1, 'MC core x', u.m)
+    mc_core_y = Field(-1, 'MC core y', u.m)
+    mc_impact = Field(-1, 'MC impact', u.m)
     n_islands = Field(-1, 'Number of image islands')
+    n_pixels = Field(-1, 'Number of pixels of cleaned images')
+    magic_stereo = Field(-1, 'True if M1 and M2 are triggered')
+
+
+class SimInfoContainer(Container):
```

Comment:
Why not using the  SimulationConfigContainer from ctapipe instead of defining a new one?  SimulationConfigContainer has more members, so you will not use them all, but I think we should try to use already defined things (especially if already in ctapipe)

Reply:
Ok, now I see that you extract `source.simulation_config`, which indeed is a `SimulationConfigContainer`, and then fill the custom container. Since the simulation configuration is not a big container (anyway just one per run), I would write directly `source.simulation_config` into the HDF5 file.

Reply:
Thanks for your comment, and indeed I agree with you, I will change it.